zrpc/internal/retry/retry.go

```diff
@@ -0,0 +1,30 @@
+package retry
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	defaultRetry = `{
+	  "name": [{"service": ""}],
+	  "retryPolicy": {
+		  "MaxAttempts": 3,
+		  "InitialBackoff": ".1s",
+		  "MaxBackoff": ".1s",
+		  "BackoffMultiplier": 1.5,
+		  "RetryableStatusCodes": ["UNAVAILABLE", "ABORTED"]
+	  }
+	}`
+)
+
+// MergeRetryPolicy merge user config and defaultRetry config
+func MergeRetryPolicy(config string) string {
```

Comment:
I think use **static structed** `ServiceConfig` is better than merge raw json string here.

Reply:
This is the official style of gRPC.
I am also considering whether to use struct.

Reply:
If we let the user specify the full service config then it is the user's responsibility to ensure that the format is correct.
But we let the user specify multiple parts separately, so our framework needs to merge the configuration to ensure the correct format. Merging raw json directly will have many boundary conditions.

Reply:
look at `retry_test.go`。

build：
`ServiceConfig `[struct Code](https://github.com/grpc/grpc-go/blob/master/service_config.go#L57)
and:
 `MethodConfig `  [struct Code](https://github.com/grpc/grpc-go/blob/master/internal/serviceconfig/serviceconfig.go#L130)
is complex.

Reply:
I see, `ServiceConfig` is really inconvenient to use in code.